src/H5VL.c

```diff
@@ -314,77 +363,28 @@ H5VLget_connector_id_by_name(const char *name)
 hid_t
 H5VLget_connector_id_by_value(H5VL_class_value_t connector_value)
 {
-    hid_t ret_value = H5I_INVALID_HID; /* Return value */
+    H5VL_connector_t *connector = NULL;
+    hid_t             ret_value = H5I_INVALID_HID; /* Return value */
 
     FUNC_ENTER_API(H5I_INVALID_HID)
 
     /* Get connector ID with this value */
-    if ((ret_value = H5VL__get_connector_id_by_value(connector_value, true)) < 0)
-        HGOTO_ERROR(H5E_VOL, H5E_CANTGET, H5I_INVALID_HID, "can't get VOL id");
-
-done:
-    FUNC_LEAVE_API(ret_value)
-} /* end H5VLget_connector_id_by_value() */
+    if (NULL == (connector = H5VL__get_connector_by_value(connector_value)))
+        HGOTO_ERROR(H5E_VOL, H5E_CANTGET, H5I_INVALID_HID, "can't get VOL connector");
 
-/*-------------------------------------------------------------------------
- * Function:    H5VLpeek_connector_id_by_name
```

Comment:
What exactly is the distinction between `H5VLget_connector_id(_by_name/value)` and `H5VLpeek_connector_id_by_name/value` such that the peek routines should be removed? All of them now return the ID to a H5VL_connector_t instead of a H5VL_class_t.

 If it has to do with the expectation that a connector can access its class under the ID, couldn't that still be achieved by looking at the `cls` field of the connector object?